icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -185,15 +433,108 @@ private static String getPerUnitFormat(ULocale locale, UnitWidth width) {
+            // FIXME(review): how does Java's fallback process differ from
+            // ICU4C's? In ICU4C we have code for explicitly falling back to
+            // unitsShort.
+        }
+    }
+
+    /**
+     * Loads and applies deriveComponent rules from CLDR's
+     * grammaticalFeatures.xml.
+     * <p>
+     * Consider a deriveComponent rule that looks like this:
+     * <pre>
+     *   &lt;deriveComponent feature="case" structure="per" value0="compound" value1="nominative"/&gt;
+     * </pre>
+     * Instantiating an instance as follows:
+     * <pre>
+     *   DerivedComponents d(loc, "case", "per");
+     * </pre>
+     * <p>
+     * Applying the rule in the XML element above, <code>d.value0("foo")</code>
+     * will be "foo", and <code>d.value1("foo")</code> will be "nominative".
+     * <p>
+     * In case of any kind of failure, value0() and value1() will simply return
+     * "".
+     */
+    private static class DerivedComponents {
+        /**
+         * Constructor.
+         */
+        DerivedComponents(ULocale locale, String feature, String structure) {
+            try {
+                ICUResourceBundle derivationsBundle =
+                    (ICUResourceBundle)UResourceBundle.getBundleInstance(ICUData.ICU_BASE_NAME,
+                                                                         "grammaticalFeatures");
+                derivationsBundle = (ICUResourceBundle)derivationsBundle.get("grammaticalData");
+                derivationsBundle = (ICUResourceBundle)derivationsBundle.get("derivations");
+
+                ICUResourceBundle stackBundle;
+                try {
+                    // TODO: use standard normal locale resolution algorithms rather than just grabbing
+                    // language:
+                    stackBundle = (ICUResourceBundle)derivationsBundle.get(locale.getLanguage());
+                } catch (MissingResourceException e) {
+                    stackBundle = (ICUResourceBundle)derivationsBundle.get("root");
+                }
+
+                stackBundle = (ICUResourceBundle)stackBundle.get("component");
+                stackBundle = (ICUResourceBundle)stackBundle.get(feature);
+                stackBundle = (ICUResourceBundle)stackBundle.get(structure);
+
+                String value = stackBundle.getString(0);
+                if (value.compareTo("compound") == 0) {
+                    this.value0 = null;
+                } else {
+                    this.value0 = value;
+                }
+
+                value = stackBundle.getString(1);
+                if (value.compareTo("compound") == 0) {
+                    this.value1 = null;
+                } else {
+                    this.value1 = value;
+                }
+            } catch (Exception e) {
+                // Fall back to uninflected.
+            }
+        }
+
+        // FIXME(review): is "getValue0" more idiomatic?
+        String value0(String compoundValue) {
```

Comment:
Would getValue0() and getValue1() be more idiomatic? (For both ICU4C and ICU4J?)

Reply:
It doesn't really matter for an internal class.  What you wrote is fine.